src/platform/api.ts

```diff
@@ -112,6 +118,13 @@ export function buildApi(
             const serverId = computeServerId(uri);
             const { controller } = await controllers.computePreferredNotebookController(notebook, serverId);
             return controller?.controller;
+        },
+        addRemoteJupyterServer: async (id: string, handle: JupyterServerUriHandle) => {
+            const connection = serviceContainer.get<JupyterConnection>(JupyterConnection);
+            const selector = serviceContainer.get<JupyterServerSelector>(JupyterServerSelector);
+            const uri = generateUriFromRemoteProvider(id, handle);
+            await connection.updateServerUri(uri);
+            await selector.setJupyterURIToRemote(uri);
```

Comment:
Currently when this code gets executed we end up switching to the new Uri, its not necessarily an add. However thats an implementation detail of Jupyter extension (which will be fixed later)